p2p/service.py

```diff
@@ -135,6 +138,7 @@ def run_task(self, awaitable: Awaitable[Any]) -> None:
                 pass
             except Exception as e:
                 self.logger.warning("Task %s finished unexpectedly: %s", awaitable, e)
+                self.logger.warning("Task failure traceback", exc_info=True)
```

Comment:
Maybe log at `DEBUG` level?

Reply:
Ah yes, that's exactly what I meant to do.